src/eigenlayer/ObolEigenLayerPodController.sol

```diff
@@ -0,0 +1,176 @@
+
+
+/// @title ObolEigenLayerFactory
+/// @author Obol
+/// @notice A factory contract for cheaply deploying ObolLidoEigenLayer.
+/// @dev The address returned should be used to as reward address for Lido
+contract ObolEigenLayerPodController {
+
+    /// @dev returned on failed call
+    error CallFailed(bytes data);
+    /// @dev If Invalid fee setup
+    error Invalid_FeeSetup();
+    /// @dev user unauthorized
+    error Unauthorized();
+    /// @dev contract already initialized
+    error AlreadyInitialized();
+
+    /// @dev Emiited on intialize
+    event Initialized(address eigenPod, address owner);
+    
+    /// -----------------------------------------------------------------------
+    /// libraries
+    /// -----------------------------------------------------------------------
+    using SafeTransferLib for address;
+    using SafeTransferLib for ERC20;
+
+    uint256 internal constant PERCENTAGE_SCALE = 1e5;
+
+    /// -----------------------------------------------------------------------
+    /// storage - immutables
+    /// -----------------------------------------------------------------------
+
+    /// @notice address of Eigenlayer delegation manager
+    /// @dev This is the address of the delegation manager transparent proxy 
+    address public immutable eigenLayerDelegationManager;
+
+    /// @notice address of EigenLayerPod Manager
+    /// @dev this is the pod manager transparent proxy
+    IEigenPodManager public immutable eigenLayerPodManager;
+
+    /// @notice address of delay withdrawal router
+    IDelayedWithdrawalRouter public immutable delayedWithdrawalRouter;
+
+    /// @notice fee address
+    address public immutable feeRecipient;
+
+    /// @notice fee share
+    uint256 public immutable feeShare;
+
+    /// -----------------------------------------------------------------------
+    /// storage
+    /// -----------------------------------------------------------------------
+
+    /// @notice address of deployed Eigen pod
+    address public eigenPod;
+
+    /// @notice address of a splitter
+    address public split;
+
+    /// @notice address of owner
+    address public owner;
+
+
+    modifier onlyOwner() {
+        if (msg.sender != owner) {
+            revert Unauthorized();
+        }
+        _;
+    }
+
+    constructor(
+        address recipient,
+        uint256 share,
+        address delegationManager,
+        address eigenPodManager,
+        address withdrawalRouter
+    ) {
+        if (recipient != address(0) && share == 0) revert Invalid_FeeSetup();
+
+        feeRecipient = recipient;
+        feeShare = share;
+        eigenLayerDelegationManager = delegationManager;
+        eigenLayerPodManager = IEigenPodManager(eigenPodManager);
+        delayedWithdrawalRouter = IDelayedWithdrawalRouter(withdrawalRouter);
+    }
+
+    /// @dev Enables contract to receive ETH
+    // defined on the clone implementation
+    // receive() external payable {}
```

Comment:
What's the story here with commenting out a payable receive? Important/not important? to keep/remove?

Reply:
it's implemented on the clone. this is meant to be documentation for integrations